src/index.ts

```diff
@@ -344,16 +372,18 @@ export function graphqlHTTP(options: Options): Middleware {
         });
       }
 
-      // Collect and apply any metadata extensions if a function was provided.
-      // https://graphql.github.io/graphql-spec/#sec-Response-Format
+      if (isAsyncIterable(executeResult)) {
```

Comment:
Since executeFn is typed to return an `AsyncIterableIterator`, it would make sense to instead use a `isAsyncIterableIterator` function. That way it is also possible to call `executeResult.return?.()` in case the request is aborted without adding ts-ignores.

Reply:
Nevermind, if you use the isAsyncIterable function th type checker already is aware of the actual type (`AsyncIterableIterator`), making it easy to call .return()